lib/modules/manager/bitrise/extract.spec.ts

```diff
@@ -0,0 +1,138 @@
+      expect(extractPackageFile('', 'bitrise.yml')).toBeNull();
+    });
+
+    it('returns valid file', () => {
+      expect(
+        extractPackageFile(
+          codeBlock`
+      workflows:
+        test:
+          steps:
+          - script@1.1.5:
+      `,
+          'bitrise.yml',
+        ),
+      ).toEqual({
+        deps: [
+          {
+            datasource: BitriseDatasource.id,
+            packageName: 'script',
+            currentValue: '1.1.5',
+            replaceString: 'script@1.1.5',
+          },
+        ],
+      });
+    });
+
+    it('returns valid file with custom default_step_lib_source', () => {
+      expect(
+        extractPackageFile(
+          codeBlock`
+      format_version: 11
+      default_step_lib_source: https://github.com/custom/steplib.git
+      project_type: android
+      app:
+        envs:
+        - MY_NAME: My Name
+      workflows:
+        test:
+          steps:
+          - script@1.1.5:
+              inputs:
+              - content: echo "Hello!"
+          - restore-cache@1.1.2:
+              foo: bar
+      `,
+          'bitrise.yml',
+        ),
+      ).toEqual({
+        deps: [
+          {
+            datasource: BitriseDatasource.id,
+            packageName: 'script',
+            currentValue: '1.1.5',
+            replaceString: 'script@1.1.5',
+            registryUrls: ['https://github.com/custom/steplib.git'],
+          },
+          {
+            datasource: BitriseDatasource.id,
+            packageName: 'restore-cache',
+            currentValue: '1.1.2',
+            replaceString: 'restore-cache@1.1.2',
+            registryUrls: ['https://github.com/custom/steplib.git'],
+          },
+        ],
+      });
+    });
+
+    it('extract git and path prefixes', () => {
+      expect(
+        extractPackageFile(
+          codeBlock`
+      workflows:
+        test:
+          steps:
+          - git::https://github.com/bitrise-io/steps-script.git@1.1.3:
+          - path::./relative/path:
+          - https://github.com/foo/bar.git::script@1:
+      `,
+          'bitrise.yml',
+        ),
+      ).toEqual({
+        deps: [
+          {
+            currentValue: '1.1.3',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/bitrise-io/steps-script.git',
+            replaceString:
+              'git::https://github.com/bitrise-io/steps-script.git@1.1.3',
+          },
+          {
+            datasource: 'bitrise',
+            packageName: 'path::./relative/path',
+            replaceString: 'path::./relative/path',
+            skipReason: 'unspecified-version',
+          },
+          {
+            currentValue: '1',
+            datasource: 'bitrise',
+            packageName: 'script',
+            registryUrls: ['https://github.com/foo/bar.git'],
+            replaceString: 'https://github.com/foo/bar.git::script@1',
+          },
+        ],
+      });
+    });
+
+    it('extract bitrise library reference', () => {
```

Comment:
```suggestion
    it('extracts a Bitrise library reference', () => {
```

Stupid question maybe, but it is possible to get more than one library reference? And if yes, is that something we should test for?

Reply:
Per step only one source can be defined. 
This can be: 
- omitted than the one defined by `default_step_lib_source` will be used
- explicitly like here
- a path delcared via `path::` prefix
- or a Git repository directly via `git::` prefix

This test case is about the second option here.

Reply:
Okay! 🙂